Pegasus/sources/particleforcegenerator.cpp

```diff
@@ -4,25 +4,25 @@
 #include <stdexcept>
 
 void pegasus::ParticleForceRegistry::add(
-    pegasus::Particle::Ptr& p, pegasus::ParticleForceGenerator::Ptr& pfg)
+    Particle::Ptr p, ParticleForceGenerator::Ptr pfg)
 {
     if (!p || !pfg) {
```

Comment:
I use a particle pointer as a key in a map, and a set of pointers to force generators as a value. It allows me to avoid duplicating particles and duplicating forces acting on this particles. I'm not sure if I can achieve the same with references.